Makefile.rules

```diff
@@ -995,6 +1002,14 @@ testbugfix: all
           ReadGapRoot( "tst/testbugfix.g" );' | $(TESTGAP) | \
             tee `date -u +dev/log/testbugfix2_%Y-%m-%d-%H-%M` )
 
+obj/libgap-test-0001.lo: $(top_srcdir)/tst/testlibgap/0001.c
+	$(QUIET_CC)$(COMPILE) $(DEPFLAGS) $(GAP_CFLAGS) $(CC_EXTRA_FLAGS) $(GAP_CPPFLAGS) -c $< -o $@
+
+testlibgap: libgap.la obj/libgap-test-0001.lo 
+	$(QUIET_LINK)$(LINK) obj/libgap-test-0001.lo libgap.la -o test-libgap
```

Comment:
I would recommend to always use `$<`, `$@` etc. instead of duplicating the names of inputs and outputs, for the usual reasons why one should avoid code duplication. So here that could become:
```
testlibgap: libgap.la obj/libgap-test-0001.lo
	$(QUIET_LINK)$(LINK) $+ -o test-libgap
	...
```

Additional thought: all other files related to this test are named `0001.FOO`, would you mind if the executable then was named `0001` (or `0001.exe` or `0001.prog` or whatever)? That would then suggest a clear why how to generalize this to more tests (i.e. just provide a list of stem names for tests; then for each such name "FOO", compile `FOO.c` into `FOO` (or `FOO.prog` or whatever), run it with input file `FOO.in` (if present), send output to `FOO.out` and diff it against `FOO.expect`.

And most importantly, it would enable us to resolve issue #1917, i.e., start providing kernel based tests for some features which are otherwise difficult to test (such as `ObjInt_Int` and its various siblings).

I can make these changes later, of course, simply want to hear what others think of the general idea.